rust/zypp-agama/zypp-agama-sys/c-layer/lib.cxx

```diff
@@ -381,6 +381,49 @@ bool is_local_url(const char *url, struct Status *status) noexcept {
   }
 }
 
+static bool package_check(Zypp *zypp, const char *tag, bool selected,
+                          Status *status) noexcept {
+  try {
+    std::string s_tag(tag);
+    if (s_tag.empty()) {
+      status->state = status->STATE_FAILED;
+      status->error = strdup("Internal Error: Package tag is empty.");
+      return false;
+    }
+
+    // look for packages
+    zypp::Capability cap(s_tag, zypp::ResKind::package);
+    zypp::sat::WhatProvides possibleProviders(cap);
+
+    for (auto iter = possibleProviders.begin(); iter != possibleProviders.end();
```

Comment:
This is a loop over *all* possible providers but it is always exited in the first iteration. I would prefer
```c++
auto iter = possibleProviders.begin();
if (iter != possibleProviders.end()) {
...
}
```
to make it clear it is intentional

Reply:
so for availability it is ok to finish at first as it means there is at least one available package....for selection, it can be any of them depending on solver and I will fix it.